docs/root/intro/arch_overview/auxiliary_config_containers.rst

```diff
@@ -0,0 +1,66 @@
+.. _arch_overview_auxiliary_config_containers:
+
+Envoy provides various auxiliary containers for config data in it's architecture,
+to help components at different layers to work correctly.
+
+
+Per Request config container
+============================
+
+StreamInfo.dynamicMetadata
+--------------------------
+
+StreamInfo.dynamicMetadata in StreamInfo allows information to be passed along
+the filter chain. A generator filter generates the data and store it in
+dynamicMetadata by an agreed key, and a consumer filter fetches the data via the agreed key.
+
+For example, HeaderToMetadataFilter can generate metadata from headers, stores
+it with key "envoy.lb", which gets consumed later by Router to make a load
+balancing decision based on metadata match result.
+
+StreamInfo.perRequestState
```

Comment:
Need to also reconcile with perConnectionState and the ongoing work from @rshriram